src/test/mina_automation/executor.ml

```diff
@@ -11,83 +11,113 @@ module DockerContext = struct
     { image : string; workdir : string; volume : string; network : string }
 end
 
-type context =
-  | Dune (* application ran from dune exec command *)
-  | Local (* application ran from _build/default folder*)
-  | Debian (* application installed from mina debian package *)
-  | Docker of DockerContext.t
-  | AutoDetect
 (* application ran inside docker container *)
 
-module Executor = struct
-  type t = { official_name : string; dune_name : string; context : context }
+module type AppPaths = sig
+  val dune_name : string
 
-  let of_context ~context ~dune_name ~official_name =
-    { context; dune_name; official_name }
-
-  let run_from_debian t ~(args : string list) ?env () =
-    Util.run_cmd_exn ?env "." t.official_name args
+  val official_name : string
+end
 
-  let run_from_dune t ~(args : string list) ?env () =
-    Util.run_cmd_exn ?env "." "dune" ([ "exec"; t.dune_name; "--" ] @ args)
+let logger = Logger.create ()
 
-  let run_from_local t ~(args : string list) ?env () =
-    Util.run_cmd_exn ?env "."
-      (Printf.sprintf "_build/default/%s" t.dune_name)
-      args
+module Make_PathFinder (P : AppPaths) = struct
+  let app_name_under_dune = P.dune_name
 
-  let built_name t = Printf.sprintf "_build/default/%s" t.dune_name
+  let official_name = P.official_name
 
   let paths =
     Option.value_map ~f:(String.split ~on:':') ~default:[] (Sys.getenv "PATH")
 
-  let exists_at_path t prefix =
-    match%bind Sys.file_exists (prefix ^ "/" ^ t.official_name) with
+  let built_name = Printf.sprintf "_build/default/%s" app_name_under_dune
+
+  let exists_at_path path prefix =
+    match%bind Sys.file_exists (prefix ^ "/" ^ path) with
     | `Yes ->
         Deferred.return (Some prefix)
     | _ ->
         Deferred.return None
 
-  let path t =
-    match%bind Sys.file_exists (built_name t) with
+  let standalone_path =
+    match%bind Sys.file_exists built_name with
     | `Yes ->
-        Deferred.return (built_name t)
+        Deferred.return (Some built_name)
     | _ -> (
-        match%bind Deferred.List.find_map ~f:(exists_at_path t) paths with
+        match%bind
+          Deferred.List.find_map ~f:(exists_at_path official_name) paths
+        with
         | Some _ ->
-            Deferred.return t.official_name
+            Deferred.return (Some official_name)
         | _ ->
-            Deferred.return t.dune_name )
```

Comment:
This branch here used to be dune_name, now it's None, what's happening?

Reply:
the name of function changed too. Now it's standalone path only applicable if we are reaching for already built app (outside dune)